e2e/recipes/recipes.go

```diff
@@ -0,0 +1,217 @@
+		},
+	}, nil
+}
+
+func generateHooks(ctx types.TestContext, rc *config.Recipe) ([]*recipe.Hook, error) {
+	var hooks []*recipe.Hook
+
+	if rc.CheckHook {
+		checkHook, err := prepareCheckHook(ctx)
+		if err != nil {
+			return hooks, err
+		}
+
+		hooks = append(hooks, checkHook)
+	}
+
+	if rc.ExecHook {
+		execHook, err := prepareExecHook(ctx)
+		if err != nil {
+			return hooks, err
+		}
+
+		hooks = append(hooks, execHook)
+	}
+
+	return hooks, nil
+}
+
+func prepareCheckHook(ctx types.TestContext) (*recipe.Hook, error) {
+	var checks []*recipe.Check
+	if checks = ctx.Workload().GetChecks(ctx.AppNamespace()); len(checks) == 0 {
+		return nil, fmt.Errorf("failed to prepare check hooks, no checks defined for workload %q",
+			ctx.Workload().GetName())
+	}
+
+	selectResource, err := validateSelectResource(ctx.Workload())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create check hook: %w", err)
+	}
+
+	return &recipe.Hook{
+		Name:           "check-hook",
+		Type:           "check",
+		Namespace:      ctx.AppNamespace(),
+		LabelSelector:  ctx.Workload().GetLabelSelector(),
+		SelectResource: selectResource,
+		Timeout:        defaultHookTimeout,
+		Chks:           checks,
+	}, nil
+}
+
+func prepareExecHook(ctx types.TestContext) (*recipe.Hook, error) {
+	var operations []*recipe.Operation
+	if operations = ctx.Workload().GetOperations(ctx.AppNamespace()); len(operations) == 0 {
+		return nil, fmt.Errorf("failed to create exec hook, no operations defined for workload %q", ctx.Workload().GetName())
+	}
+
+	selectResource, err := validateSelectResource(ctx.Workload())
+	if err != nil {
+		return nil, fmt.Errorf("failed to create exec hook %w", err)
+	}
+
+	return &recipe.Hook{
+		Name:           "exec-hook",
+		Type:           "exec",
+		Namespace:      ctx.AppNamespace(),
+		NameSelector:   ctx.Workload().GetAppName(),
+		SelectResource: selectResource,
+		Timeout:        defaultHookTimeout,
+		Ops:            operations,
+	}, nil
+}
+
+func validateSelectResource(w types.Workload) (string, error) {
+	selectResource := w.GetSelectResource()
+	if _, ok := supportedSelectResources[selectResource]; !ok {
+		return selectResource, fmt.Errorf("selectResource %q is not supported for workload %q (choose from %v)",
+			selectResource, w.GetName(), slices.Collect(maps.Keys(supportedSelectResources)))
+	}
+
+	return selectResource, nil
+}
+
+func generateGroups(ctx types.TestContext, appNamespace string) []*recipe.Group {
+	return []*recipe.Group{
+		{
+			Name:      defaultGroupName,
+			Type:      resource,
+			BackupRef: defaultGroupName,
+			IncludedNamespaces: []string{
+				appNamespace,
+			},
+			LabelSelector: ctx.Workload().GetLabelSelector(),
+		},
+	}
+}
+
+func generateWorkflows(recipeSpec *config.Recipe, hooks []*recipe.Hook) []*recipe.Workflow {
+	backup := &recipe.Workflow{Name: backup}
+	restore := &recipe.Workflow{Name: restore}
+
+	checkHooks, execHooks := collectHooks(hooks)
```

Comment:
The new function is very simple - good! but we don't validate the hooks argument with the configuration. This code will silently return workflow without a exec hook if there is no such hook the the hook slice, while the config may have execHook: true.